### PR TITLE
Wide subcategories

### PR DESCRIPTION
**Pull request by nvarner**
Defines wide subcategories (`HomPropertyOver`) for #54. Includes some basic examples, and incorporates some preliminary feedback from @stschaef.

### CODE REVIEW

File: Cubical/Categories/Displayed/Constructions/HomPropertyOver.agda [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    ℓC ℓC' ℓP : Level
+
+open Category
+
+module _
+  (C : Category ℓC ℓC') (P : ∀ {x y} → C [ x , y ] → Type ℓP)
+  (Pprop : ∀ {x y} (f : C [ x , y ]) → isProp (P f))
+  (Pid : ∀ {x} → P (id C {x}))
+  (Pcomp : ∀ {x y z} (f : C [ x , y ]) (g : C [ y , z ]) → P f → P g → P (_⋆_ C f g))
+  where
```
Comment (maxsnew) [inline Cubical/Categories/Displayed/Constructions/HomPropertyOver.agda:22]: I think there's so many fields here that we should make this a struct `HomPropertyOver` like we do with `StructureOver`.

**Review by maxsnew — CHANGES_REQUESTED** [review]
Looks great but I think we should make it a struct like we do with StructureOver

File: Cubical/Categories/Displayed/Constructions/HomPropertyOver.agda (hunk 1 above)
Comment (nvarner) [reply Cubical/Categories/Displayed/Constructions/HomPropertyOver.agda]: I've changed `HomPropertyOver` to a record and created a defined to displayed categories.

Comment (nvarner) [reply Cubical/Categories/Displayed/Constructions/HomPropertyOver.agda]: I've updated these names to match, except for `Hom[_]`. It caused a parser ambiguity between `Hom[_,_]` and `Hom[ (_,_) ]`. I used `Hom[_][-,-]` instead, which is the most reasonable idea I've had.

**Pull request merged**